### DISCUSSION TITLE
Gruppieren von Einzelverbrauchern

### DISCUSSION THREAD

**Discussion by Zeppelin-PV** [solectrus/solectrus#3764] [Q&A] [answered]
nachdem auch ich den Weg gefunden habe, wie aus HomeAssistant Daten zu holen sind, - DANKE dafür Georg
mache ich mir Gedanken über eine Gruppierung um so die benutzerdefinierten Verbraucher im Zaum zu halten
So stelle ich mir vor, dass unter IT die Fritzbox, der Switch, das NAS, Laptop etc. zusammengeführt werden
Unter Multimedia TVs, Radios, etc
und für mich speziell Verbraucher, die im direkten Zusammenhang stehen mit PFLEGE, 
wie können einzelne Verbraucher zusammen addiert werden bevor sie unter dem jeweiligen Begriff in der Datenbank verschwinden ?
Ist das für andere auch interessant ??? - oder gibt es hierzu womöglich schon eine Lösung ???

**Comment by gereons** [comment, marked as answer]
In https://github.com/orgs/solectrus/discussions/3392#discussioncomment-10465272 gibt's eine Idee wie man mehrere Shellys so verbinden kann dass einer von denen die Aggregation übernimmt. 

Alternativ bräuchte man einen Server der diesen Job erledigen kann (mehrere MQTT-Subscriptions aufmachen, die Daten nach benutzerdefinierten Formeln aggregieren und das Ergebnis dann wiederum per MQTT publishen), aber den gibt's auf den ersten Blick nicht einfach so fertig zum 'runterladen.

**Reply by Zeppelin-PV** [reply]
@gereons Danke für den Tip
zwischenzeitlich bin ich auf folgende Lösung gestoßen
da bei mir fast alle Power-Messungen über HomeAssistant laufen mit ZigBee Mess-Steckdosen kann ich dort die Summe einzelner Verbraucher bilden und dort auch schon namentlich so benennen wie die Gruppe lauten soll. Dann müsste dies auch über den Mosquitto Broker an Solectrus zu übertragen sein. Muss ich mir mal anschauen. 
... welche Sonderzeichen und Umlaute sind im MQTT Datenverkehr erlaubt ???

**Reply by gereons** [reply]
MQTT-Topics sollten mit Umlauten klarkommen, da sie UTF-8-Strings sind. Vermeiden solltest du nur die Zeichen "+", "#", "\*", ">" und "$".

**Comment by ledermann** [comment]
Eine Lösungsmöglichkeit wäre vermutlich eine Gruppenverwaltung direkt in SOLECTRUS. Eine Gruppe wäre die Zusammenfassung mehrerer Verbraucher, deren Messwerte bei Bedarf aufsummiert werden.

Man würde (in den Einstellungen) Gruppen anlegen können und einzelne Verbraucher diesen zuordnen. Auf der Hausverbrauchs-Seite würden dann entweder die einzelnen Verbraucher angezeigt werden (wie bisher) oder die Gruppen.

Das MQTT-Setup würde nicht tangiert werden und auch in InfluxDB würde nichts zusätzlich gespeichert werden.

**Reply by Zeppelin-PV** [reply]
hallo Georg, das hört sich auch ganz gut an
aber egal ob gruppiert oder nicht, 10 Verbraucher sind zu wenig für mich, wie schon unter #3749 mitgeteilt
